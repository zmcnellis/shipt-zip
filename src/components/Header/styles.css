/* Navbar --------------------------------------------------------------------*/
header {
  width: 100%;
  height: 85px;
  margin: 0;
  box-sizing: border-box;
  box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.5);
}

nav {
  box-sizing: border-box;
  margin: 0 auto;
  max-width: 1080px;
  width: 100%;
  height: 100%;
  padding: 0 24px 0 24px;
  font-size: 2em;
  color: #333;
  background-color: #fff;
}

.nav-right {
  float: right;
}

.nav-right .button {
  display: inline-block;
  margin: 17px 0;
}

nav > .logo {
  height: 50px;
  margin: 17px 0 18px 0;
}
