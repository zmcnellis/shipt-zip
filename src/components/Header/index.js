import React, { Component } from 'react';
import logo from '../../images/logo.svg';
import './styles.css';

class Header extends Component {
  render() {
    return (
      <header>
        <nav>
          <img className="logo" src={logo} alt="logo" />
          <div className="nav-right">
            <a className="button button-primary" href="#">Sign Up</a>
          </div>
        </nav>
      </header>
    );
  }
}

export default Header;
