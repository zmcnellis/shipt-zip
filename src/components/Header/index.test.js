import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Header from '.';

describe('Header', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Header/>
    );
    expect(output).toMatchSnapshot();
  });

  it('should have semantic tags', () => {
    const output = shallow(
      <Header/>
    );
    expect(output.find('header').exists()).toBeTruthy();
    expect(output.find('nav').exists()).toBeTruthy();
    expect(output.find('img.logo').exists()).toBeTruthy();
  });
});
