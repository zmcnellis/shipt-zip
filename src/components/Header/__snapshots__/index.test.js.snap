// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav>
        <img
                alt="logo"
                className="logo"
                src="logo.svg"
        />
        <div
                className="nav-right"
        >
                <a
                        className="button button-primary"
                >
                        Sign Up
                </a>
        </div>
</nav>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="logo"
            src="logo.svg"
/>,
          <div
            className="nav-right"
>
            <a
                        className="button button-primary"
            >
                        Sign Up
            </a>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "logo",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="button button-primary"
>
              Sign Up
</a>,
            "className": "nav-right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "button button-primary",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "a",
          },
          "type": "div",
        },
      ],
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav>
          <img
                    alt="logo"
                    className="logo"
                    src="logo.svg"
          />
          <div
                    className="nav-right"
          >
                    <a
                              className="button button-primary"
                    >
                              Sign Up
                    </a>
          </div>
</nav>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="logo"
              src="logo.svg"
/>,
            <div
              className="nav-right"
>
              <a
                            className="button button-primary"
              >
                            Sign Up
              </a>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="button button-primary"
>
                Sign Up
</a>,
              "className": "nav-right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "className": "button button-primary",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "a",
            },
            "type": "div",
          },
        ],
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
